### ISSUE TITLE
[Bug 2.3] Use-release not working

### ISSUE THREAD

**Issue by ghilainm** [mojohaus/versions-maven-plugin#79]
I have a BOM file (only dependencyManagement therefore and no dependencies).

I apply the following command 'mvn clean org.codehaus.mojo:versions-maven-plugin:2.3-SNAPSHOT:use-releases' and none of the dependencies in SNAPSHOT are replaced with the corresponding release (trust me they exist in maven repository).

I also tried without success to force the property processDependencyManagement (which anyway should default to true).

I have already observe such kind of behavior with your plugin and I believe it is a bug.

EDIT: This does not work either with dependencies

**Comment by ghilainm** [comment]
According to me this will never work (maven 3.3.1 used, I have made the fix and provided a pull request.)

**Comment by wcurrie** [comment]
Seems like this is the same as (and fixes) #72. Seems to have been reported on stackoverflow without any clear resolution too: http://stackoverflow.com/questions/11193070/maven-versions-plug-in-skipping-update-check

**Comment by tao-zhang** [comment]
Same issue here. Always fails when you try to upgrade dependency version, if your repository only has release enabled or  only snapshot enabled. If both are enabled, it will work. But that's not what we want.

**Comment by gingasteve** [comment]
I've just started investigating the use of this plugin to manage our dependency version numbers. This bug will almost certainly prevent me from using this plugin.

**Comment by traderjoe1** [comment]
I´m waiting for bugfix :-)

**Comment by stephenjohnwilliams** [comment]
Is this problem ever going to be fixed? Is this plugin still being maintained I wonder?

**Comment by khmarbaise** [comment]
Unfortunately the patch of ghilainm was no PR...I have taken the liberty to take the code into the plugin. Thanks for that.

**Comment by ghilainm** [comment]
What does PR mean?

**Comment by khmarbaise** [comment]
Ah sorry Pull Request...

**Comment by ghilainm** [comment]
I  did a pull request fixing several issues ☺.

**Comment by khmarbaise** [comment]
I've seen that you made a pull request which contains several fixes which is hard to follow...it is better having a pull request referencing a single issue which makes simpler to follow and simpler to integrate the pull requests and fixes into the plugin...Nevertheless thanks for you support...I will take a deeper look...
